### ISSUE TITLE
Change column width on secure spreadsheets?

### ISSUE THREAD

**Issue by mikesierra2** [standardnotes/forum#1514]
I have a note set with the Secure Spreadsheets Editor. I would like to set individual column widths, but this is not an option in the editor. If this is indeed possible, what tricks do I use? If it is not available yet, is this a future feature?

I am using Android and Windows 10, neither allow column width changes. When I click where I would click to change the width in any other spreadsheet, the border gets a bit thicker indicating I might be able too drag it, but it is not drag-able in any way. 

Thanks for any help

**Comment by stefan-yas** [comment]
Hey @mikesierra2, you should be able to successfully resize a column by holding the edge of the column header and dragging. It's not currently possible to set pixel-perfect column widths, but it may happen in the future, if the editor which Secure Spreadsheets is derived from introduces such a feature.

![image](https://user-images.githubusercontent.com/52251483/149733522-d95b999b-28cf-49da-8a6d-d21120a41b5e.png)

**Comment by mikesierra2** [comment]
That is not currently working on either on the app installed on my Galaxy Note 20, or the Standard Notes installed on Windows 10. Tried creating multiple notes with Secure Spreadsheets, none allow column resize on either OS. On both OS, the resizing bar embiggens just like your image, but it cannot be dragged left or right. I have used both the stylus and my finger on the Note 20.

Is this a bug, or is there a setting to allow this feature? I am paid up with the older 5 year Standard Notes Extended license.

**Comment by stefan-yas** [comment]
Your subscription status should not matter as long as you have Secure Spreadsheets installed as an editor. Mind attaching a screen recording from your Windows 10 device attempting to resize the column width? Is your Windows 10 device a touch device?

**Comment by mikesierra2** [comment]
If it helps, I am using v3.8.21 on the Windows 10 machine. Yes, it is touch, it is a Surface Book 3. But generally I am using external monitors with a mouse. I just now uninstalled the Spreadsheets addin, and reinstalled it. It is version 1.4.2.

This is what I see when I hover over the are where I should be able to resize columns. It is not draggable. 
![image](https://user-images.githubusercontent.com/80733025/150248169-6fa9659e-c5ac-4252-8b04-f8de321c27d9.png)

I think you wanted a recording rather than a still. I don't have any software that can do a recording. If I click the mouse button on the image above, then the mouse icon moves, but the column separator stays where it is.

**Comment by mikesierra2** [comment]
Also, I can change the column width by editing the underlying note definition. Change editor to the normal note editor, and hack away. This is a note that I just created, added some text to column 1 row 1, and then changed the default column width by changing the number in the definition.

{"activeSheet":"Sheet1","sheets":[{"name":"Sheet1","rows":[{"index":0,"height":46,"cells":[{"value":"test test test test test test","wrap":true,"index":0}]}],"columns":[],"selection":"A1:A1","activeCell":"A1:A1","frozenRows":0,"frozenColumns":0,"showGridLines":true,"gridLinesColor":null,"mergedCells":[],"hyperlinks":[],"defaultCellStyle":{"fontFamily":"Arial","fontSize":12},"drawings":[]}],"names":[],"columnWidth":164,"rowHeight":20,"rows":75,"columns":26}

Note that there is a row height in that. I cannot do that with the UI by dragging the row separator. I did that by entering a long text, and turning on word wrap. I then can edit the row height by changing that number.

**Comment by mikesierra2** [comment]
So, playing about with the underlying spreadsheet definition in the plain editor, I can now set the column width. I added the column definition below, adding this to the existing empty column definition:
[{"index":0,"width":126}],



{"activeSheet":"Sheet1","sheets":[{"name":"Sheet1","rows":[{"index":0,"cells":[{"value":"testing testing testing","index":0}]}],"columns":[{"index":0,"width":126}],"selection":"A2:A2","activeCell":"A2:A2","frozenRows":0,"frozenColumns":0,"showGridLines":true,"gridLinesColor":null,"mergedCells":[],"hyperlinks":[],"defaultCellStyle":{"fontFamily":"Arial","fontSize":12},"drawings":[]}],"names":[],"columnWidth":64,"rowHeight":20,"rows":75,"columns":26}


It still would be nice to get the UI drag and drop working.

Oh, I should say, adding that column definition did not make the UI column width drag and drop feature work.

**Comment by mikesierra2** [comment]
Using the pen on the Surface Book 3 touch screen still does not allow column resize.
I saw that there was an app update on Windows, so updated Standard Notes to 3.9.15, still no resizable columns.
Having set the column width as noted above, editing the defintion, I opened the note on Android, and the column was wider as expected. Still no ability to drag the column width in Android.

**Comment by stefan-yas** [comment]
@mikesierra2 Are you using Windows 10 in tablet mode? If so, can you try controlling the width with your mouse after disabling touch mode?

**Comment by mikesierra2** [comment]
No, Win 10 is not in tablet mode. It is docked to two external monitors. I have tried both the touch screen and the mouse to control the width and neither work. And on the Note 20 (Android), neither the stylus, nor the finger work. I am intrigued by this, as it does not work on two different devices, with two different OS, and multiple HID options. Assuming it works for other people, I cannot think of anything that would affect that broad range of devices.

**Comment by mikesierra2** [comment]
I just discovered standardnotes.com/demo. I created a Secure Spreadsheets note in that site on three different browsers on win 10, and two different browsers on Android, all were unable to set the column width by dragging the column separator.

**Comment by stefan-yas** [comment]
Indeed, that is weird. Here's a recording of me using Secure Spreadsheets on standardnotes.com/demo to control the column width. I had a hunch it may have something to do with your Windows 10 device being touch-enabled, but I'm not sure what could be the cause right now.

You wouldn't be able to change the column width easily on mobile devices using touch. I believe it's only possible with a mouse.

https://user-images.githubusercontent.com/52251483/150961990-141b21a6-e21c-41d5-8e03-cb92c193056b.mp4

**Comment by mikesierra2** [comment]
OK, I know what is causing the problem, so you may be able to test it at your end. I have a Surface Book 3. This allows me to dock the screen to the keyboard either facing forwards like a laptop, or backwards so it becomes a thick tablet. I always use the backwards option. I then have it docked with a Bluetooth mouse and a USB keyboard, and two monitors. If the screen and keyboard are docked in the laptop mode, then Secure spreadsheets column resize works. This is regardless of the laptop being undocked or docked to the monitors keyboard and mouse.

But while the laptop keyboard and monitor are docked in the reverse orientation, the Secure Spreadsheets column resize does not work. Note that at all times, I have "Tablet mode" turned off. The bluetooth mouse can change the column width when docked in the laptop mode, but not when screen is docked reverse. Happy to help with debugging/testing if that helps.

Also, you mentioned that secure spreadsheets column resize is not working with Android phones. Is this expected in the future? It is actually on the Android Phone that I would most often be using Secure Spreadsheets.

Thanks
David

**Comment by stefan-yas** [comment]
I see, thanks for making light of this. The Secure Spreadsheets editor is derived from [KendoUI](https://demos.telerik.com/kendo-ui/spreadsheet/index), so it's up to the maintainer of the editor to fix the behavior of this functionality with touch-enabled, dockable devices. I've submitted a report now of this behavior to the maintainers in hopes they will fix the functionality.

> Also, you mentioned that secure spreadsheets column resize is not working with Android phones. Is this expected in the future? It is actually on the Android Phone that I would most often be using Secure Spreadsheets.

Perhaps this may be possible in the future, by tapping the column header which could open a menu where you could find this function.

**Comment by siopaosmuggler** [comment]
I am also unable to resize the columns by clicking and dragging the columns as shown in Stefan's video, both in browser (tried Firefox and Edge) and in the Windows App (v. 3.20.0)

Edit: added more details
